### PR TITLE
[Console] Added SymfonyStyle::definitionList() and SymfonyStyle::horizontalTable()

### LINKED ISSUE #12286: [Console] Added support for definition list and horizontal table
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#32742
| PR author(s) | @lyrixx
| Merged in    | 4.4

### PR DESCRIPTION
**Pull request by lyrixx**
fixes #12286

### CODE REVIEW

**Review by OskarStark — APPROVED** [review]
But please add a versionadded directive below each example.

@javiereguiluz what do you think of adding screenshots? 🤔

**Comment by javiereguiluz** [comment]
@OskarStark no screenshots please (because they are very hard to maintain), but in the future we'll add `.. code-block:: terminal` examples to show the result of each method.

**Comment by javiereguiluz** [comment]
Thanks Grégoire! I added the `versionadded` directives while merging.

**Comment by lyrixx** [comment]
Thanks @javiereguiluz

File: console/style.rst [hunk 1]
```diff
@@ -140,6 +140,31 @@ Content Methods
...
+            ['Header 1', 'Header 2'],
+            [
+                ['Cell 1-1', 'Cell 1-2'],
+                ['Cell 2-1', 'Cell 2-2'],
+                ['Cell 3-1', 'Cell 3-2'],
+            ]
+        );
+
+:method:`Symfony\\Component\\Console\\Style\\SymfonyStyle::definitionList`
+    It displays the given arguments as compact table horizontally where the key::
```
Comment (xabbuh) [inline console/style.rst]: @lyrixx What does the "where the key" part of the sentence refer to?

Comment (lyrixx) [reply console/style.rst]: Oups, I don't know what I wanted to say. It's a type that should be removed.
Do you want me to do a PR?

Comment (xabbuh) [reply console/style.rst]: That would be great. :) Or if you leave a comment here about what sentence should be placed there, I can do that too.

Comment (lyrixx) [reply console/style.rst]: Oh, it's already fixed by Javier in a5e1d81ce5ac8c463f782456baa55b5e40af389d :heart:

Comment (xabbuh) [reply console/style.rst]: 😆 I should have checked. Thanks @javiereguiluz! 👍

**Pull request merged**